### PR TITLE
UCP/PROTO: Add protocol performance calculation and stubs for tag/eager

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucp/tag/eager_single.c [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#include <ucs/sys/string.h>
+
+
+static ucs_status_t
+ucp_proto_eager_short_init(const ucp_proto_init_params_t *init_params)
+{
+    const ucp_proto_select_param_t *select_param = init_params->select_param;
+    ucp_proto_single_init_params_t params = {
+        .super.super         = *init_params,
+        .super.latency       = -150e-9,
```
Comment (brminich) [inline src/ucp/tag/eager_single.c:25]: is this avoiding fetching data?

File: src/ucp/proto/proto_common.c [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                if (!(md_attr->cap.flags & UCT_MD_FLAG_REG) ||
+                    !(md_attr->cap.reg_mem_types & UCS_BIT(select_param->mem_type))) {
+                    ucs_trace("lane[%d]: no reg of mem type %s", lane,
+                              ucs_memory_type_names[select_param->mem_type]);
+                    continue;
+                }
+            } else {
+                /* Memory domain which does not requite a registration for zero
+                 * copy operation must be able to access the relevant memory type
+                 * TODO UCT should expose a bitmap of accessible memory types
```
[the remark this answers is not in the document]
Comment (yosefe) [reply src/ucp/proto/proto_common.c:144]: yes, but in the future want to advertise that an MD can access more than one type

File: src/ucp/proto/proto_common.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+typedef struct {
+    ucp_proto_init_params_t super;
+    double                  latency;       /* protocol added latency */
+    double                  overhead;      /* protocol overhead */
+    size_t                  cfg_thresh;    /* user-configured threshold */
+    size_t                  fragsz_offset; /* offset of maximal fragment
+                                              size in uct_iface_attr_t */
+    size_t                  hdr_size;      /* header size on first lane */
+    unsigned                flags;         /* see ucp_proto_common_init_flags_t */
+} ucp_proto_common_init_params_t;
```
[the remark this answers is not in the document]
Comment (yosefe) [reply src/ucp/proto/proto_common.h:30]: these parameters are used not only for performance calculation, but also lane initialization, for example by `ucp_proto_single_init`

File: src/ucp/tag/eager_single.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/tag/eager_single.c:25]: yes

**Pull request merged**